### PR TITLE
Allow to specify the inputs.allowed_setup on benchmark workflow dispatch

### PR DESCRIPTION
**Pull request by fcostaoliveira**
This PR allows for more precise benchmark retriggers. By default it won't change the current flow. but if you specify the allowed_setup input it will only run that setup related flows and skip all others.

### CODE REVIEW

**Review by raz-mon — APPROVED** [review]

File: .github/workflows/benchmark-runner.yml [hunk 1]
```diff
@@ -143,7 +150,7 @@ jobs:
       benchmark_runner_group_total: ${{ strategy.job-total }}
 
   benchmark-search-oss-standalone-profiler:
-    if: inputs.profiler
+    if: ${{ inputs.extended && (inputs.allowed_setup == '' || inputs.allowed_setup == 'standalone') }}
```
Comment (nafraf) [inline .github/workflows/benchmark-runner.yml:153]: Shouldn't use `inputs.profiler` ?
```suggestion
    if: ${{ inputs.profiler && (inputs.allowed_setup == '' || inputs.allowed_setup == 'standalone') }}
```

File: .github/workflows/benchmark-runner.yml [hunk 2]
```diff
@@ -163,7 +170,7 @@ jobs:
 
 
   benchmark-vecsim-oss-standalone-profiler:
-    if: inputs.profiler
+    if: ${{ inputs.extended && (inputs.allowed_setup == '' || inputs.allowed_setup == 'standalone') }}
```
Comment (nafraf) [inline .github/workflows/benchmark-runner.yml:173]:
```suggestion
    if: ${{ inputs.profiler && (inputs.allowed_setup == '' || inputs.allowed_setup == 'standalone') }}
```

File: .github/workflows/benchmark-runner.yml [hunk 3]
```diff
@@ -21,9 +21,14 @@ on:
       profiler:
         type: boolean
         default: false
+      allowed_setup:
+        type: string
+        description: 'if empty allows all setups. if not, will only trigger for a specific setup'
+        default: ""
```
[the remark this answers is not in the document]
Comment (fcostaoliveira) [reply .github/workflows/benchmark-runner.yml:31]: addressed :)

**Review by raz-mon — APPROVED** [review]

**Review by nafraf — APPROVED** [review]

**Comment by raz-mon** [comment]
\backport

**Pull request merged**